src/core/component.js

```diff
@@ -324,9 +326,8 @@ Component.prototype = {
 
     // For oldData, pass empty object to multiple-prop schemas or object single-prop schema.
     // Pass undefined to rest of types.
-    initialOldData = this.isObjectBased ? this.objectPool.use() : undefined;
+    initialOldData = this.isObjectBased ? emptyInitialOldData : undefined;
```

Comment:
Same here. My understanding is that the `use` function calls `clearObject`. You mention 

> The pool however makes no guarantees about the state of the object. It might very well not be an empty object

What Am I missing?